src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Ball {
+    private String color;
+    private int number;
+
+    public void setColor() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        this.color = colorSupplier.getRandomColor();
+    }
+
+    public void setNumber() {
+        Random random = new Random();
+        this.number = random.nextInt(100);
+    }
```

Comment:
setters should not have any logic inside them, they should only set a field of our object that it gets in the parameters, `Lottery` class should be responsible for creating a ball with random parameters by using our `ColorSupplier` class in a [composition-manner](https://www.infoworld.com/article/3409071/java-challenger-7-debugging-java-inheritance.html#:~:text=Inheritance%20and%20composition%20are%20two,the%20sum%20of%20its%20parts.) structure

Reply:
also, don't forget about making magic numbers (like this `100`) a constants with a proper naming